src/session.rs

```diff
@@ -156,31 +156,47 @@ impl Session {
         }
     }
 
-    /// Set a new key for the one or both of the initiator-egress and responder-egress symmetric ciphers.
+    /// Generates a new key for the the initiator-egress symmetric cipher according to
+    /// Section 4.2 of the Noise Specification. Synchronizing timing of rekey between
+    /// initiator and responder is the responsibility of the application, as described in
+    /// Section 11.3 of the Noise Specification.
     ///
     /// # Errors
     ///
     /// Will result in `SnowError::State` if not in transport mode.
     #[must_use]
-    pub fn rekey(&mut self, initiator: Option<&[u8]>, responder: Option<&[u8]>) -> Result<(), SnowError> {
```

Comment:
I don't think we should totally clobber the old "manual" rekeying process in case other applications are using different rekey mechanisms. Maybe instead we can rename it to `rekey_manual(...)` or something else that similarly signals "you're on your own".

Reply:
fixed in 3ce7be18dfcc